providers/redfish/firmware.go

```diff
@@ -127,6 +134,11 @@ func (c *Conn) FirmwareInstall(ctx context.Context, component, applyAt string, f
 	return taskID, nil
 }
 
+type multipartPayload struct {
+	updateParameters io.Reader
```

Comment:
So I went looking up the calls stack and now I'm wondering, why does this need to be an io.Reader at all? Its always going to be a `[]byte` from `json.Marshal` in https://github.com/bmc-toolbox/bmclib/blob/a1b87e2ff47f5c05022a6cb8fddc6811e13d9a89/providers/redfish/firmware.go#L79-L87. I don't see why we want to make it a Reader and have to have all the complications of tee'ing and resetting. If its kept as a []byte then down below in multipartPayloadSize p
it'd be something like:

```go
if _, err = part.Write(payload.updateParameters); err != nil {
		return 0, body, err
}
```
right?

Reply:
thanks for feedback there, [08172f2](https://github.com/bmc-toolbox/bmclib/pull/345/commits/08172f2d91032140fb93567da8046fbcf5b76b0a)